ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -8,9 +8,56 @@
 
 import Foundation
 
+public protocol MutexType {
+	func lock()
+	func unlock()
+}
+
+public final class RecursiveLock: MutexType {
```

Comment:
Should we use a pthread recursive mutex here, if the Objective-C runtime dispatch cost is/was a concern for `Atomic`?

Otherwise, instead of reinventing the wheel, `Locking`(`NSLocking`) available in both Objective-C and Swift Foundation sounds a better option.